src/__tests__/subscription.test.ts

```diff
@@ -0,0 +1,1057 @@
+    const schema = new GraphQLSchema({
+      query: DummyQueryType,
+      subscription: new GraphQLObjectType({
+        name: "Subscription",
+        fields: {
+          foo: {
+            type: GraphQLString,
+            subscribe: () => "test"
+          }
+        }
+      })
+    });
+
+    const document = parse("subscription { foo }");
+
+    (await expectPromise(subscribe({ schema, document }))).toRejectWith(
+      'Subscription field must return Async Iterable. Received: "test".'
+    );
+  });
+
+  it("resolves to an error for subscription resolver errors", async () => {
+    async function subscribeWithFn(subscribeFn: () => any) {
+      const schema = new GraphQLSchema({
+        query: DummyQueryType,
+        subscription: new GraphQLObjectType({
+          name: "Subscription",
+          fields: {
+            foo: { type: GraphQLString, subscribe: subscribeFn }
+          }
+        })
+      });
+      const document = parse("subscription { foo }");
+      const result = await subscribe({ schema, document });
+
+      return result;
+    }
+
+    const expectedResult = {
+      errors: [
+        {
+          message: "test error",
+          locations: [{ line: 1, column: 16 }],
+          path: ["foo"]
+        }
+      ]
+    };
+
+    expect(
+      // Returning an error
+      await subscribeWithFn(() => new Error("test error"))
+    ).toEqual(expectedResult);
+
+    expect(
+      // Throwing an error
+      await subscribeWithFn(() => {
+        throw new Error("test error");
+      })
+    ).toEqual(expectedResult);
+
+    expect(
+      // Resolving to an error
+      await subscribeWithFn(() => Promise.resolve(new Error("test error")))
+    ).toEqual(expectedResult);
+
+    expect(
+      // Rejecting with an error
+      await subscribeWithFn(() => Promise.reject(new Error("test error")))
+    ).toEqual(expectedResult);
+  });
+
+  it("resolves to an error if variables were wrong type", async () => {
+    const schema = new GraphQLSchema({
+      query: DummyQueryType,
+      subscription: new GraphQLObjectType({
+        name: "Subscription",
+        fields: {
+          foo: {
+            type: GraphQLString,
+            args: { arg: { type: GraphQLInt } }
+          }
+        }
+      })
+    });
+
+    const variableValues = { arg: "meow" };
+    const document = parse(`
+      subscription ($arg: Int) {
+        foo(arg: $arg)
+      }
+    `);
+
+    // If we receive variables that cannot be coerced correctly, subscribe() will
+    // resolve to an ExecutionResult that contains an informative error description.
+    const result = await subscribe({ schema, document, variableValues });
+    expect(result).toEqual({
+      errors: [
+        {
+          message:
+            // DIFF: 'Variable "$arg" got invalid value "meow"; Int cannot represent non-integer value: "meow"',
+            'Variable "$arg" got invalid value "meow"; Expected type Int; Int cannot represent non-integer value: "meow"',
```

Comment:
I think, if it is not too breaking, we should change this message to stay up to date with `graphql-js`. Wdyt?

Reply:
Yes, we can change it.